### PR TITLE
Add Addon content to tutorial (knobs)

### LINKED ISSUE #21: Add chapter on integrating add-ons using knobs as an example
Many people would benefit from a tutorial on add-on integration
Many people would love knobs (very widely utilized add-on)

I'm down to work on content. Just wanted to make sure this is something that's desired.

### PR DESCRIPTION
**Pull request by kylemh**
Sorry it took so long! I wanted to make sure I'm going in the right direction.

_Feel free to edit this!_

[ ] - Translate to Spanish (I'll ask a coworker about this tomorrow)
[ ] - Verify integration with website (include screenshot) [BLOCKED BY #39]
[ ] - Merge code example edits [here](https://github.com/hichroma/learnstorybook-code/pull/3) and add commit ref
[ ] - Update screenshots of example Storybook UI that occur after this step in the tutorial to avoid confusing users. 

Resolves #21

### CODE REVIEW

**Comment by kylemh** [comment]
Sounds great @icarlossz 👍

**Comment by tmeasday** [comment]
Hi @kylemh, sorry to be slow to respond, have been out of town. This seems great, I wil take a proper look soon

**Comment by kylemh** [comment]
@tmeasday LOL no rush at all... my turn around was like 3 months 😂 

Please don't stress at all about it.

File: content/react/en/addons.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+First, import the `withKnobs` decorator and the `object` knob type to `Task.stories.js`:
+
+```javascript
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { action } from '@storybook/addon-actions';
+import { withKnobs, object } from '@storybook/addon-knobs/react';
+```
+
+Next, within the stories of `Task`, pass `withKnobs` as a parameter to the `addDectorator()` function:
```
Comment (tmeasday) [inline content/react/en/addons.md]: Small typo (`addDectorator)`

File: content/react/en/addons.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+<div class="aside">
+<strong>Looking for a list of potential addons?</strong>
+<br/>
+😍 You can seen the list of officially-supported and strongly-supported community addons <a href="https://storybook.js.org/addons/addon-gallery/">here</a>.
+</div>
+
+We could write forever about configuring and using addons for all of your particular use-cases. For now, let's work towards integrating one of the most popular addons within Storybook's ecosystem: [knobs](https://github.com/storybooks/storybook/tree/v3.3.15/addons/knobs).
+
+<div class="aside">
+<strong>⚠ Version 3.3.15 of Knobs ⚠</strong>
```
Comment (tmeasday) [inline content/react/en/addons.md]: Was there a reason for this particular version?

Comment (kylemh) [reply content/react/en/addons.md]: I felt weird about this one. I just wanted to be crystal clear about matching up with the dependencies outlined in the example code base. The installation and usage look to be the same on master, so I'll revert the version-specifying content.

**Comment by kylemh** [comment]
@icarlossz I think this is ready for translation 👍

**Comment by tmeasday** [comment]
Hi @kylemh -- sorry about the delay in merging this. I think maybe the end of this chapter could be a bit more definitive -- I think beginners who are reading a tutorial like this would probably prefer to have a concrete suggestion of what to do next.

I was thinking maybe we should just extend your example of text overflowing and follow through the process of adding a new story and fixing the issue. What do you think?

I am happy to make the change but I thought I should give you the chance first; I was thinking it could be something like:

```markdown
## Using Knobs To Find Edge-Cases

Additionally, with easy access to editing passed data to a component, QA Engineers or preventating UI Engineers can now push their component to the limit! As an example, what happens to `Task` if our list item has *MASSIVE* string?

![Oh no!](/addon-knobs-demo-edge-case.png)

Thanks to quickly being able to try different inputs to a component we can find and fix such problems with relative ease! Let's fix the issue with overflowing by adding a style to `Task.js`:

```jsx
// This is the input for our task title. In practice we would probably update the styles for this element  
// but for this tutorial, let's fix the problem with an inline style:
<input type="text" value={title} readOnly={true} placeholder="Input title" style={{ 'text-overflow': 'ellipsis' }} />
```

That's better!

### Adding a new story to avoid regressions

Of course we can always reproduce this problem by entering the same input into the knobs, but it's to write a fixed story for this input. This will increase your regression testing and clearly outline the limits of the component(s) to the rest of your team.

Let's add a story for the long text case in `Task.stories.js`:

```js
const longTitle = `This task's name is absurdly large. In fact, I think if I keep going I might end up with content overflow. What will happen? The star that represents a pinned task could have text overlapping. The text could cut-off abruptly when it reaches the star. I hope not`;

storiesOf('Task', module)
  .add('default', () => <Task task={task} {...actions} />)
  .add('pinned', () => <Task task={{ ...task, state: 'TASK_PINNED' }} {...actions} />)
  .add('archived', () => <Task task={{ ...task, state: 'TASK_ARCHIVED' }} {...actions} />)
  .add('long title', () => <Task task={{ ...task, title: longTitle }} {...actions} />);
```

**Comment by kylemh** [comment]
Thanks so much for giving me the opportunity. I'll tackle it ASAP.

**Comment by kylemh** [comment]
Sorry about the delay on this one. Hoping to get to the new content Friday on a flight ✈️  👍

**Comment by kylemh** [comment]
@xxsz I think you're in the clear to translate ❤️ 
@tmeasday I like this content so much more - good work 🙇.

**Comment by tmeasday** [comment]
Great! Published

**Pull request merged**